api/config/esgateway/config_request.go

```diff
@@ -89,6 +91,20 @@ func (c *ConfigRequest) Validate() error {
 // to start the service.
 func (c *ConfigRequest) PrepareSystemConfig(creds *ac.TLSCredentials) (ac.PreparedSystemConfig, error) {
 	c.V1.Sys.Tls = creds
+
+	enableSystemNameServer := c.GetV1().GetSys().GetNgx().GetMain().GetResolvers().GetEnableSystemNameservers().GetValue()
+	nameServers := c.GetV1().GetSys().GetNgx().GetMain().GetResolvers().GetNameservers()
+	if len(nameServers) == 0 && enableSystemNameServer {
+		c.V1.Sys.Ngx.Main.Resolvers.NameserversString = getSystemResolvers()
+	} else {
+		if (nameServers != nil) && (len(nameServers) > 0) {
```

Comment:
Similar to the previous site, this could be 

```suggestion
		if len(nameServers) > 0 {
```